src/handle.rs

```diff
@@ -271,63 +271,62 @@ impl WlcOutput {
     /// For example tiling wms, may want to use this to keep their tiling order separated
     /// from floating order.
     /// This handles `wlc_output_get_views` and `wlc_output_get_mutable_views`.
-    pub fn get_views(&self) -> Vec<WlcView> {
+    pub fn get_views(self) -> Vec<WlcView> {
         unsafe {
             let mut out_memb: libc::size_t = 0;
             let views = wlc_output_get_views(self.0, &mut out_memb);
-            if views.is_null() {
-                return Vec::new();
-            }
             let mut result = Vec::with_capacity(out_memb);
-
-            for index in (0 as isize) .. (out_memb as isize) {
-                  result.push(WlcView(*(views.offset(index))));
+            if !views.is_null() {
+                for index in (0 as isize) .. (out_memb as isize) {
+                      result.push(WlcView(*views.offset(index)));
+                }
             }
-            return result;
+            result
         }
     }
 
     /// Gets the mask of this output
-    pub fn get_mask(&self) -> u32 {
+    pub fn get_mask(self) -> u32 {
         unsafe { wlc_output_get_mask(self.0) }
     }
 
     /// Sets the mask for this output
-    pub fn set_mask(&self, mask: u32) {
+    pub fn set_mask(self, mask: u32) {
         unsafe { wlc_output_set_mask(self.0, mask) }
     }
 
     /// # Deprecated
     /// This function is equivalent to simply calling get_views
-    pub fn get_mutable_views(&self) -> Vec<WlcView> {
+    pub fn get_mutable_views(self) -> Vec<WlcView> {
         self.get_views()
     }
 
     /// Attempts to set the views of a given output.
     ///
     /// Returns success if operation succeeded. An error will be returned
     /// if something went wrong or if wlc isn't running.
-    pub fn set_views(&self, views: &mut Vec<&WlcView>) -> Result<(), &'static str> {
-            let view_len = views.len() as libc::size_t;
-            let view_vals: Vec<uintptr_t> = views.into_iter().map(|v| v.0).collect();
-            let const_views = view_vals.as_ptr();
+    pub fn set_views(self, views: &[WlcView]) -> Result<(), &'static str> {
+        let view_len = views.len() as libc::size_t;
+        let const_views = views.as_ptr() as *const uintptr_t;
+
         unsafe {
-            match wlc_output_set_views(self.0, const_views, view_len) {
+            let res = match wlc_output_set_views(self.0, const_views, view_len) {
```

Comment:
NB this res = x; return x; pair is leftover from when I was instrumenting println! to debug